File: app/src/main/java/com/nruge/iceinfo/model/OsmModels.kt
package com.nruge.iceinfo.model

data class TrackInfo(
    val maxSpeed: Int? = null,
    val electrified: String? = null,
    val voltage: Int? = null,
    val tracks: Int? = null,
    val usage: String? = null
)

enum class RailFeatureType { TUNNEL, BRIDGE, STATION, HALT }

data class RailFeature(
    val name: String,
    val type: RailFeatureType,
    val distanceKm: Double
)

data class OsmTrackData(
    val trackInfo: TrackInfo = TrackInfo(),
    val features: List<RailFeature> = emptyList(),
    val isLoading: Boolean = false,
    val error: String? = null
)

File: ios/ICEInfo/Models/OsmModels.swift
import Foundation

struct TrackInfo: Codable, Sendable {
    var maxSpeed: Int? = nil
    var electrified: String? = nil
    var voltage: Int? = nil
    var tracks: Int? = nil
    var usage: String? = nil
}

enum RailFeatureType: String, Codable, Sendable {
    case tunnel = "TUNNEL"
    case bridge = "BRIDGE"
    case station = "STATION"
    case halt = "HALT"
}

struct RailFeature: Codable, Sendable, Identifiable {
    var id: String { name + type.rawValue }
    var name: String
    var type: RailFeatureType
    var distanceKm: Double
}

struct OsmTrackData: Codable, Sendable {
    var trackInfo: TrackInfo = TrackInfo()
    var features: [RailFeature] = []
    var isLoading: Bool = false
    var error: String? = nil
}
